src/main/java/ee/tuleva/onboarding/mandate/event/AfterMandateSignedEvent.java

```diff
@@ -21,15 +21,8 @@ public AfterMandateSignedEvent(Object source, User user, Mandate mandate, Locale
     this.locale = locale;
   }
 
-  public static AfterMandateSignedEvent newInstance(
-      Object source, User user, Mandate mandate, Locale locale) {
-    if (mandate.getPillar() == 2) {
-      return new SecondPillarAfterMandateSignedEvent(source, user, mandate, locale);
-    } else if (mandate.getPillar() == 3) {
-      return new ThirdPillarAfterMandateSignedEvent(source, user, mandate, locale);
-    } else {
-      throw new IllegalArgumentException("Event for the pillar type is not available");
-    }
+  public Integer getPillar() {
```

Comment:
We could add a `getSuggestedPillar()` method here that does the logic of:
`return mandate.getPillar() == 2 ? 3 : 2;`